sktime/transformers/series_as_features/signature_based/_augmentations.py

```diff
@@ -0,0 +1,148 @@
+import torch
+from sklearn.pipeline import Pipeline
+from sktime.transformers.series_as_features.base import \
+    BaseSeriesAsFeaturesTransformer
+
+
+def get_augmentation_pipeline(aug_list):
+    """Buids an sklearn pipeline of augmentations from a list of strings.
+
+    Parameters
+    ----------
+    aug_list: list of strings, A list of strings that determine the
+        augmentations to apply, and in which order to apply them (the first
+        string will be applied first). Possible augmentation strings are
+        ['leadlag', 'ir', 'addtime', 'cumsum', 'basepoint']
+
+    Returns
+    -------
+    sklearn.Pipeline
+        The transforms, in order, as an sklearn pipeline.
+
+    Examples
+    --------
+    >>> get_augmentation_pipeline(['leadlag', 'ir', 'addtime'])
+    Pipeline([
+        ('leadlag', LeadLag()),
+        ('ir', InvisibilityReset()),
+        ('addtime', AddTime())
+    ])
+    """
+    # Assertions
+    types = [tuple, list, None, str]
+    assert any([type(aug_list) == t for t in types]), (
+        "`aug_list` must be one of {}. Got {}.".format(types, type(aug_list))
+    )
+    aug_list = [aug_list] if isinstance(aug_list, str) else aug_list
+
+    # Dictionary of augmentations
+    AUGMENTATIONS = {
+        'leadlag': LeadLag(),
+        'ir': InvisibilityReset(),
+        'addtime': AddTime(),
+        'cumsum': CumulativeSum(),
+        'basepoint': BasePoint()
+    }
+
+    pipeline = Pipeline([
+        (tfm_str, AUGMENTATIONS[tfm_str]) for tfm_str in aug_list
+    ])
+
+    return pipeline
+
+
+class AddTime(BaseSeriesAsFeaturesTransformer):
+    """Add time component to each path.
+
+    For a path of shape [B, L, C] this adds a time channel to be placed at the
+    first index. The time channel will be of length L and scaled to exist in
+    [0, 1].
+    """
+    def transform(self, data):
+        # Batch and length dim
+        B, L = data.shape[0], data.shape[1]
+
+        # Time scaled to 0, 1
+        time_scaled = torch.linspace(0, 1, L).repeat(B, 1).view(B, L, 1)
+
+        return torch.cat((time_scaled, data), 2)
+
+
+class InvisibilityReset(BaseSeriesAsFeaturesTransformer):
+    """Adds an 'invisibility-reset' dimension to the path. This adds
+    sensitivity to translation.
+
+    Introduced by Yang et al.: https://arxiv.org/pdf/1707.03993.pdf
+    """
+    def transform(self, X):
+        # Batch, length, channels
+        B, L, C = X.shape[0], X.shape[1], X.shape[2]
+
+        # Add in a dimension of ones
+        X_pendim = torch.cat((torch.ones(B, L, 1), X), 2)
+
+        # Add pen down to 0
+        pen_down = X_pendim[:, [-1], :]
+        pen_down[:, :, 0] = 0
+        X_pendown = torch.cat((X_pendim, pen_down), 1)
+
+        # Add home
+        home = torch.zeros(B, 1, C + 1)
+        X_penoff = torch.cat((X_pendown, home), 1)
+
+        return X_penoff
+
+
+class LeadLag(BaseSeriesAsFeaturesTransformer):
```

Comment:
same here, I'd suggest making this a private class, unless it's meant to be used directly by users, otherwise perhaps a somewhat more descriptive name